### PR TITLE
C++ API: torch::nn::Softmin

### PR DESCRIPTION
**Pull request by nuka137**
Add torch::nn::Softmin module and functional support for the C++ API.

Related Issue: #25883 

Reviewer: @yf225

### CODE REVIEW

**Comment by yf225** [comment]
Thanks a lot for the great work! We will need some fixes as suggested in https://github.com/pytorch/pytorch/pull/27446.

**Comment by nuka137** [comment]
@yf225 

Fixed all issues discussed in #27446, and rebased to newest master branch.
Could you review this PR again?

File: torch/csrc/api/include/torch/nn/options/activation.h [hunk 1]
```diff
@@ -61,5 +61,15 @@ struct TORCH_API SoftmaxOptions {
   TORCH_ARG(int64_t, dim);
 };
 
+// ============================================================================
+
+/// Options for the Softmin functional and module.
+struct TORCH_API SoftminOptions {
+  SoftminOptions(int64_t dim);
+
+  // Dimension along which Softmin will be computed.
```
Comment (yf225) [inline torch/csrc/api/include/torch/nn/options/activation.h]: nit: we should change `//` to `///` so that the comment will show up in the C++ API docs (my apologies for missing it for the `Softmax` module 😞)

Comment (nuka137) [reply torch/csrc/api/include/torch/nn/options/activation.h]: @yf225 

If you don't mind, I'll also fix `Softmax` module in this PR.
What do you think about this idea?

Comment (yf225) [reply torch/csrc/api/include/torch/nn/options/activation.h]: Yes that would be awesome, thanks so much for your help @nuka137!

Comment (nuka137) [reply torch/csrc/api/include/torch/nn/options/activation.h]: Fixed them.

**Comment by nuka137** [comment]
@yf225 

I fixed the comments on `SoftminOptions` and `SoftmaxOptions`.
Could you check them?
And I noticed that there are conflicts, do I need some work against them?

**Review by yf225 — APPROVED** [review]
@nuka137 The changes look awesome, thanks so much for your help! I will fix the merge conflicts :D
